examples/morpheusvm/genesis/genesis.go

```diff
@@ -9,90 +9,48 @@ import (
 	"fmt"
 
 	"github.com/ava-labs/avalanchego/trace"
-	"github.com/ava-labs/avalanchego/x/merkledb"
 
 	"github.com/ava-labs/hypersdk/codec"
 	"github.com/ava-labs/hypersdk/examples/morpheusvm/consts"
 	"github.com/ava-labs/hypersdk/examples/morpheusvm/storage"
-	"github.com/ava-labs/hypersdk/fees"
 	"github.com/ava-labs/hypersdk/state"
 	"github.com/ava-labs/hypersdk/vm"
 
 	smath "github.com/ava-labs/avalanchego/utils/math"
-	hconsts "github.com/ava-labs/hypersdk/consts"
 )
 
-var _ vm.Genesis = (*Genesis)(nil)
+var (
+	_ vm.Genesis                    = (*Genesis)(nil)
+	_ vm.RuleFactory[*vm.BaseRules] = (*RuleFactory)(nil)
+)
+
+type RuleFactory struct {
+	unchangingRules *vm.BaseRules
+}
+
+func (r *RuleFactory) GetRules(_ int64) *vm.BaseRules {
+	return r.unchangingRules
+}
 
 type CustomAllocation struct {
 	Address string `json:"address"` // bech32 address
 	Balance uint64 `json:"balance"`
 }
 
 type Genesis struct {
-	// State Parameters
-	StateBranchFactor merkledb.BranchFactor `json:"stateBranchFactor"`
-
-	// Chain Parameters
-	MinBlockGap      int64 `json:"minBlockGap"`      // ms
-	MinEmptyBlockGap int64 `json:"minEmptyBlockGap"` // ms
-
-	// Chain Fee Parameters
-	MinUnitPrice               fees.Dimensions `json:"minUnitPrice"`
-	UnitPriceChangeDenominator fees.Dimensions `json:"unitPriceChangeDenominator"`
-	WindowTargetUnits          fees.Dimensions `json:"windowTargetUnits"` // 10s
-	MaxBlockUnits              fees.Dimensions `json:"maxBlockUnits"`     // must be possible to reach before block too large
-
-	// Tx Parameters
-	ValidityWindow      int64 `json:"validityWindow"` // ms
-	MaxActionsPerTx     uint8 `json:"maxActionsPerTx"`
-	MaxOutputsPerAction uint8 `json:"maxOutputsPerAction"`
-
-	// Tx Fee Parameters
-	BaseComputeUnits          uint64 `json:"baseUnits"`
-	StorageKeyReadUnits       uint64 `json:"storageKeyReadUnits"`
-	StorageValueReadUnits     uint64 `json:"storageValueReadUnits"` // per chunk
-	StorageKeyAllocateUnits   uint64 `json:"storageKeyAllocateUnits"`
-	StorageValueAllocateUnits uint64 `json:"storageValueAllocateUnits"` // per chunk
-	StorageKeyWriteUnits      uint64 `json:"storageKeyWriteUnits"`
-	StorageValueWriteUnits    uint64 `json:"storageValueWriteUnits"` // per chunk
+	*vm.BaseRules
 
 	// Allocates
 	CustomAllocation []*CustomAllocation `json:"customAllocation"`
 }
 
+func (g *Genesis) GetRulesFactory() vm.RuleFactory[*vm.BaseRules] {
+	return &RuleFactory{g.BaseRules}
+}
+
```

Comment:
Can this be implemented in `vm.BaseRules` so that we don't need to here and it's up to the VM to wrap and override it if desired?